src/main/java/com/htisolutions/poolref/services/ResetPasswordService.java

```diff
@@ -0,0 +1,100 @@
+package com.htisolutions.poolref.services;
+
+import com.htisolutions.poolref.entities.SecurityQuestion;
+import com.htisolutions.poolref.entities.SecurityQuestionDao;
+import com.htisolutions.poolref.entities.User;
+import com.htisolutions.poolref.entities.UserDao;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.authentication.UsernamePasswordAuthenticationToken;
+import org.springframework.security.authentication.dao.DaoAuthenticationProvider;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.stereotype.Service;
+
+
+@Service
+public class ResetPasswordService {
+    private UserDao userDao;
+    private SecurityQuestionDao securityQuestionDao;
+    private DaoAuthenticationProvider authenticationProvider;
+    private User user;
+    private boolean answeredQuestion;
+
+    @Autowired
+    ResetPasswordService(UserDao userDao, SecurityQuestionDao securityQuestionDao, DaoAuthenticationProvider authenticationProvider) {
+        this.userDao = userDao;
+        this.securityQuestionDao = securityQuestionDao;
+        this.authenticationProvider = authenticationProvider;
+        user = null;
+        answeredQuestion = false;
+    }
+
+    public boolean canAnswerQuestion(){
+        return (user != null);
+    }
+    public boolean canResetPassword(){
+        return (user != null && answeredQuestion);
+    }
+    public Boolean validName(String nickname){
+        Iterable <User> userList = userDao.findAll();
+        for(User user: userList){
+            if (user.getNickname().equals(nickname)){
+                if(user.getSecurity_question_id() != null) {
+                    this.user = user;
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+    public void autologin(String password) {
+        UsernamePasswordAuthenticationToken token = new UsernamePasswordAuthenticationToken(user.getNickname(), password);
+        Authentication authentication = authenticationProvider.authenticate(token);
+        SecurityContextHolder.getContext().setAuthentication(authentication );
+    }
+    public boolean validPassword(String password, String confirm){
```

Comment:
This is a bad naming convention, because it is doing more than just validating the password, it is also saving to the Db